src/tss2-fapi/ifapi_helpers.c

```diff
@@ -2517,7 +2517,19 @@ ifapi_get_curl_buffer(unsigned char * url, unsigned char ** buffer,
         goto out_global_cleanup;
     }
 
+#ifdef CURLU_ALLOW_SPACE
+    CURLU *urlp = curl_url();
+    rc = curl_url_set(urlp, CURLUPART_URL, url, CURLU_ALLOW_SPACE | CURLU_URLENCODE);
+    if (rc != CURLE_OK) {
+        LOG_ERROR("curl_url_set for CURUPART_URL failed: %s",
+                  curl_easy_strerror(rc));
```

Comment:
If this fails, we need a `curl_url_cleanup(urlp)` or AFAICT we will leak as we never transferred ownership to the curl pointer. Their is a chance we hit this case when curl_url() returns a NULL pointer, which curl_url_cleanup is safe to call with a NULL pointer.